File: app/src/main/java/be/icapps/template/ApplicationProvider.java
package be.icapps.template;

import android.app.Application;
import android.content.pm.PackageManager;
import android.util.Log;

import com.google.android.gms.analytics.GoogleAnalytics;
import com.google.android.gms.analytics.Tracker;
import com.icapps.crashreporter.CompoundCrashReporter;
import com.icapps.crashreporter.CrashLog;

import crashreporter.crittercism.CrittercismCrashReporter;
import crashreporter.googleanalytics.GoogleAnalyticsCrashReporter;

public class ApplicationProvider extends Application {

	private Tracker mTracker;

	@Override
	public void onCreate() {
		super.onCreate();

		if (!BuildConfig.DEBUG) {
			final CompoundCrashReporter compoundCrashReporter = new CompoundCrashReporter();

			final CrittercismCrashReporter.Config critConfig = new CrittercismCrashReporter.Config();
			critConfig.includeVersionCode = true;
			critConfig.logNetworkCalls = false;

			compoundCrashReporter.addCrashLogger(new CrittercismCrashReporter(this, getString(R.string.crittercism_id), critConfig));
			compoundCrashReporter.addCrashLogger(new GoogleAnalyticsCrashReporter(getTracker()));

			CrashLog.initialize(compoundCrashReporter);
			CrashLog.leaveBreadcrumb("App Started");
		}
	}

	public Tracker getTracker() {
		if (mTracker == null) {
			final GoogleAnalytics analytics = GoogleAnalytics.getInstance(this);
			analytics.enableAutoActivityReports(this);
			mTracker = analytics.newTracker(R.xml.analytics);
			mTracker.enableAdvertisingIdCollection(true);
			mTracker.enableAutoActivityTracking(true);
			try {
				mTracker.setAppVersion(getPackageManager().getPackageInfo(getPackageName(), 0).versionName);
			} catch (PackageManager.NameNotFoundException e) {
				Log.d("Analytics Tracker", e.getMessage());
			}
			mTracker.setAppName(getString(R.string.app_name));
			mTracker.enableExceptionReporting(true);
		}
		return mTracker;
	}
}

File: app/src/main/kotlin/be/icapps/template/ApplicationProvider.kt
package be.icapps.template

import android.app.Application
import android.content.pm.PackageManager
import android.util.Log
import com.github.salomonbrys.kodein.Kodein
import com.github.salomonbrys.kodein.KodeinAware
import com.github.salomonbrys.kodein.bind
import com.github.salomonbrys.kodein.singleton
import com.google.android.gms.analytics.GoogleAnalytics
import com.google.android.gms.analytics.Tracker
import com.icapps.crashreporter.CompoundCrashReporter
import com.icapps.crashreporter.CrashLog
import crashreporter.crashlytics.CrashlyticsCrashReporter
import crashreporter.googleanalytics.GoogleAnalyticsCrashReporter

class ApplicationProvider : Application(), KodeinAware {

    lateinit override var kodein: Kodein

    private val mTracker: Tracker by lazy {
        val analytics = GoogleAnalytics.getInstance(this)
        analytics.enableAutoActivityReports(this)
        analytics.newTracker(R.xml.analytics)
                .apply {
                    enableAutoActivityTracking(true)
                    setAppName(getString(R.string.app_name))
                    enableExceptionReporting(true)

                    try {
                        setAppVersion(packageManager.getPackageInfo(packageName, 0).versionName)
                    } catch (e: PackageManager.NameNotFoundException) {
                        Log.d("Analytics Tracker", e.message)
                    }
                }
    }

    override fun onCreate() {
        super.onCreate()

        kodein = Kodein {
            bind<Application>() with singleton { this@ApplicationProvider }
        }


        if (!BuildConfig.DEBUG) {
            val compoundCrashReporter = CompoundCrashReporter()

            compoundCrashReporter.addCrashLogger(CrashlyticsCrashReporter(this))
            compoundCrashReporter.addCrashLogger(GoogleAnalyticsCrashReporter(mTracker))

            CrashLog.initialize(compoundCrashReporter)
            CrashLog.leaveBreadcrumb("App Created")
        }
    }

}